clients/java: describe job worker polling and backoff (#185)
camunda-cloud/camunda-cloud-documentation: docs/product-manuals/clients/java-client/job-worker.md

---
id: job-worker
title: "Job worker"
---

## Related Resources

- [Job worker basics](/product-manuals/concepts/job-workers.md)

## The java client's job worker

The java client provides a job worker that takes care of polling for available jobs and allows users to focus on writing code to handle the activated jobs.

Generally, the job worker:

> - polls for jobs before it runs out

Reviewer: Before it runs out...of jobs? ...on its family after going for a pack of cigarette at the corner store? :thinking:

- waits for `pollInterval` milliseconds (and polls again) when there are no jobs available
- applies a backoff strategy on polling errors

## Example usage

- [Open a job worker](../java-client-examples/job-worker-open.md)

## Backoff configuration

> When a poll fails with an error response: the job worker waits for some time, afterwhich it polls again for available jobs.

Reviewer: ```suggestion
When a poll fails with an error response: the job worker waits for some time, after which it polls again for available jobs.
```

---
id: job-worker
title: "Job worker"
---

> ## Related Resources

Reviewer 2: ```suggestion
## Related resources
```

- [Job worker basics](/product-manuals/concepts/job-workers.md)

> ## The java client's job worker

Reviewer 2: ```suggestion
## The Java client's job worker

The Java client provides a job worker that takes care of polling for available jobs and allows you to focus on writing code to handle the activated jobs.
```
